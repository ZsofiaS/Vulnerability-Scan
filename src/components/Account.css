.accountContainer {
  position: relative;
}
.accountDropdown {
  position: absolute;
}
