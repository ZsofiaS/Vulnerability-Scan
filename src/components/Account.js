import React, { useState } from 'react';

const Account = (props) => {

  const [hidden, setHidden] = useState(true);

  return (
    <div className='accountContainer'>
      <span>
        <button onClick={() => setHidden(!hidden)}>
          Account
        </button>
      </span>
      { !hidden && (
        <div className='accountDropdown'>
          <p>{props.name}</p>
          <p>{props.email}</p>
        </div>
      )}
    </div>
  )
}

export default Account;
