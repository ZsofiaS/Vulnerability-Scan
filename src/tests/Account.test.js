import React from 'react';
import { shallow, mount } from 'enzyme';
import Account from '../components/App';
import toJson from 'enzyme-to-json';

const user = {
    name: 'Joe',
    email: 'joe@joe.com'
}

describe('renders Account component', () => {
  it('renders without crashing', () => {
    shallow(<Account />);
  })
});

describe('it passes props', () => {
  const account = mount(<Account name={user.name} />);
  it('accepts props', () => {
    expect(account.props().name).toEqual('Joe');
  })
})

describe('snapshot', () => {
  it('Account snapshot', () => {
    const tree = shallow(<Account />);
    expect(toJson(tree)).toMatchSnapshot();
  })
})
