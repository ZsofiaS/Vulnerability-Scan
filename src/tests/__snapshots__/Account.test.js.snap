// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot Account snapshot 1`] = `
<div
  className="App"
>
  <TopBar />
  <Summary />
  <Details />
</div>
`;
